### DISCUSSION TITLE
Private npm registry stopped working

### DISCUSSION THREAD

**Discussion by segevfiner** [renovatebot/renovate#32909] [Request Help]
### How are you running Renovate?

A Mend.io-hosted app

### Please tell us more about your question or problem

We migrated our npm private registry secret as instructed by the renovate dashboard, and we are using the following config:
```json5
  "npmrcMerge": true,
  "npmToken": "{{ secrets.GITHUB_COM_TOKEN }}",
  "npmrc": "@swimmio:registry=https://npm.pkg.github.com/\n//npm.pkg.github.com/:_authToken=${NPM_TOKEN}"
```

With this `.npmrc` already existing in the repo:
```
engine-strict=true
@swimmio:registry=https://npm.pkg.github.com
```

But renovate seems to go to the default npm registry instead of the private GitHub Packages npm registry as configured and I'm not sure why or what's wrong...

This config used to work, and I'm not sure what broke it...

There seems to be a myriad of different ways of configuring auth and the docs about it seem kinda messy and could use reorganizing to point to the recommended way (Referring to https://docs.renovatebot.com/getting-started/private-packages/#npm). In particular I'm not sure what renovate prefers between hostRules and `npmrc`, and whether hostRules propogates into `npmrc` at all (As renovate seems to be invoking `pnpm` by itself for updating the lock file which would require the token to be there)

### Logs (if relevant)

<details><summary>Logs</summary>

```

DEBUG: GET https://registry.npmjs.org/@swimmio%2Fprivate-package = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=404 retryCount=0, duration=6288)
DEBUG: Failed to look up npm package @swimmio/private-package
{
  "dependency": "@swimmio/private-package"
  "packageFile": "apps/swimm/package.json"
}

```

</details>

**Comment by segevfiner** [comment]
Looks like this is screwing this config up... https://github.com/renovatebot/renovate/blob/b69416ce1745f67c9fc1d149738e2f52feb4f732/lib/config/massage.ts#L24-L26 I'll need to try other things :(

**Reply by segevfiner** [reply]
Also renovate is ignoring any repo `.npmrc` even with no `npmrc` set in config. So it seems I'll have to set the `npmrc` in config. Now to see if the automatic auth works, or if I also need to set up `hostRules` or the `authToken` in that `npmrc`.

**Reply by segevfiner** [reply]
Nope. automatic auth to GitHub packages is not working. No npm.pkg.github.com in the generated hostRules in the log at all... Trying `hostRules` now as described here: https://docs.renovatebot.com/getting-started/private-packages/#automatically-authenticate-for-npm-package-stored-in-private-github-npm-repository

**Reply by segevfiner** [reply]
The example linked above seems to have worked. The private registry docs really need some updating... (Maybe setting a custom npmrc in the config with a `{{ secrets.WHATEVER }}` would have also worked). But in particular `npmToken` which is mentioned in the docs seems primarily intended for the main npm registry and cause the issue due to the massage config routine.

**Comment by viceice** [comment]
@rarkins @secustor I think we should deprecate `npmToken`. eg drop from docs samples and mark as docs deprecation only for now.

so new users don't start using it and get these troubles.
